app/controllers/public_profiles_controller.rb

```diff
@@ -0,0 +1,14 @@
+class PublicProfilesController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    @developer = developer
```

Comment:
We also need to authorize that the current user owns the developer profile. We can create a new Pundit policy in `app/policies` or use the existing one but pass in `:update?` as a custom method. Your call!

Reply:
I have updated existing developer_policy.rb file with new method share_profile? and authorizing in controller. Not sure about passing update? custom method mentioned in above comment.. Can you confirm if current implementation looks fine?